omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/ConceptSearchHandler1_8.java

```diff
@@ -162,6 +162,18 @@ public PageableResult search(RequestContext context) throws ResponseException {
 			throw new InvalidSearchException("Invalid searchType: " + searchType
 			        + ". Allowed values: \"equals\" and \"fuzzy\"");
 		}
+
+		// getting concepts by classUuid
+		if (conceptClass != null) {
+			ConceptClass responseConceptClass = conceptService.getConceptClassByUuid(conceptClass);
+			if (responseConceptClass != null) {
+
+				List<Concept> concept = conceptService.getConceptsByClass(responseConceptClass);
+				concepts.addAll(concept);
+				return new NeedsPaging<Concept>(concepts, context);
```

Comment:
The ticket description says that we should be able to filter concepts by class, in addition to other parameters. What you have implemented filters by class and leaves out the rest of the parameters. In other words, `AND` the given parameters. e.g `WHERE name=ASP && searchType=fuzzy && class=8d490dfc-c2cc-11de-8d13-0010c6dffd0f`. Does this make sense?

Reply:
> The ticket description says that we should be able to filter concepts by class, in addition to other parameters. What you have implemented filters by class and leaves out the rest of the parameters. In other words, `AND` the given parameters. e.g `WHERE name=ASP AND searchType=fuzzy AND class=8d490dfc-c2cc-11de-8d13-0010c6dffd0f`. Does this make sense?

hello @dkayiwa i may not have gotten your comment correctly but i have just test this endpoint `/openmrs/ws/rest/v1/concept?name=ASP&searchType=fuzzy&class=8d490dfc-c2cc-11de-8d13-0010c6dffd0f` and  it works fine unless i missed something in your comment please enlighted me 
 
because according to @mks-d  this endpont `/openmrs/ws/rest/v1/concept?class=8d490dfc-c2cc-11de-8d13-0010c6dffd0f` was the issue, that if the name parameter was dropped the class parameter would not work so i included a method that filters the concept 

are you suggesting that i include more  method to cater for the name as well?
please let me know

Reply:
because i think thats the only one missing 
for in my understnding the searchType is already catered for with this method 
`// If there's class parameter in query
		if ("fuzzy".equals(searchType)) {
			List<Locale> locales = new ArrayList<Locale>(LocaleUtility.getLocalesInOrder());
			List<ConceptClass> classes = null;
			ConceptClass responseConceptClass = conceptService.getConceptClassByUuid(conceptClass);
			
			if (responseConceptClass != null) {
				classes = Arrays.asList(responseConceptClass);
			}
			
			List<ConceptSearchResult> searchResults = conceptService.getConcepts(name, locales, context.getIncludeAll(),
			    classes, null, null, null, null, context.getStartIndex(), context.getLimit());
			List<Concept> results = new ArrayList<Concept>(searchResults.size());
			for (ConceptSearchResult csr : searchResults) {
				results.add(csr.getConcept());
			}
			return new NeedsPaging<Concept>(results, context);
		} else if (searchType == null || "equals".equals(searchType)) {
			
			if (name != null) {
				Concept concept = conceptService.getConceptByName(name);
				concepts.add(concept);
				if (concept != null) {
					boolean isPreferredOrFullySpecified = false;
					for (ConceptName conceptname : concept.getNames()) {
						if (conceptname.getName().equalsIgnoreCase(name)
						        && (conceptname.isPreferred() || conceptname.isFullySpecifiedName())) {
							isPreferredOrFullySpecified = true;
							break;
						}
					}
					if (!isPreferredOrFullySpecified) {
						throw new APIException(
						        "The concept name should be either a fully specified or locale preferred name");
					}
					
					return new NeedsPaging<Concept>(concepts, context);
				} else {
					return new EmptySearchResult();
				}
			}
		} else {
			throw new InvalidSearchException("Invalid searchType: " + searchType
			        + ". Allowed values: \"equals\" and \"fuzzy\"");
		}`

Reply:
plus if a drop the class parameter and just run http://localhost:8080/openmrs/ws/rest/v1/concept?name=ASP&searchType=fuzzy it still works

Reply:
According to your implementation, if the class `8d490dfc-c2cc-11de-8d13-0010c6dffd0f` has concepts with names `ASP` and `XYZ`, it will return both of them simply because they have that class. Yet you are required to filter even the name within that class, according to the search parameters.

Reply:
if am to get you correctly is that i should i write a method for that filters concepts by name such that if we are to call `/openmrs/ws/rest/v1/concept?name=ASP&class=8d490dfc-c2cc-11de-8d13-0010c6dffd0f` it should filter by name and class, and also filter comcepts by only name also

is this so @dkayiwa

Reply:
that is to say `/openmrs/ws/rest/v1/concept?name=ASP`

Reply:
or you would want me to append the current implementation to add the name parameter as a filter

Reply:
Let me put it differently, can you write three tests where one makes the call `/openmrs/ws/rest/v1/concept?name=ASP&class=8d490dfc-c2cc-11de-8d13-0010c6dffd0f` and the other `/openmrs/ws/rest/v1/concept?name=XYZ&class=8d490dfc-c2cc-11de-8d13-0010c6dffd0f`, then lastly `/openmrs/ws/rest/v1/concept?class=8d490dfc-c2cc-11de-8d13-0010c6dffd0f`

Reply:
i got it now
let me try to make some changes

Reply:
hello @dkayiwa 
i have managed to make some changes 
we can now search concept by both name and class or class alone 
however iam faced by a challenge it can only work if you spell out the fullnames of the concept that is ` /openmrs/ws/rest/v1/concept?name=caffeine&class=8d490dfc-c2cc-11de-8d13-0010c6dffd0f` but abbreviations or call them acro-names ` /openmrs/ws/rest/v1/concept?name=CAF&class=8d490dfc-c2cc-11de-8d13-0010c6dffd0f` to make a request is challenging

i kindly request further guidance on this 
thank

Reply:
hello @dkayiwa 
as i explored further i release we maybe dont really need to write a new method for the classfilter logic we may just need to edit `if ("fuzzy".equals(searchType))` to something like this `if ("fuzzy".equals(searchType) || conceptClass != null)` becuase the code needed is the same and it corrects all the failures of the ticket 

However we run into a challage of testing the if else part where the results  should return an empty string 
i have commented out my implementation for now and the two test failures then complied the code ran sdk and tested out everything, it worked fine except for the empty result and the exception

Reply:
i still for your comment @dkayiwa

Reply:
what would be your on this please